htlcswitch/switch.go

```diff
@@ -2464,11 +2500,13 @@ func (s *Switch) getLinkByMapping(pkt *htlcPacket) (ChannelLink, error) {
 	chanID := pkt.outgoingChanID
 	aliasID := s.cfg.IsAlias(chanID)
 
+	_, haveAliasMapping := s.aliasToReal[chanID]
```

Comment:
This should work, think we need to update `failAliasUpdate` as well

Reply:
Great catch, added as well.